modules/JsonExporter/src/main/java/uk/ac/ox/oii/jsonexporter/JSONExporter.java

```diff
@@ -0,0 +1,277 @@
+/*
+Copyright (C) 2015  Scott A. Hale
+Website: http://www.scotthale.net/
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/license
+ */
+package uk.ac.ox.oii.jsonexporter;
+
+import com.google.gson.Gson;
+import java.awt.Color;
+import java.io.FileNotFoundException;
+import java.io.Writer;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import org.gephi.graph.api.Column;
+import org.gephi.graph.api.Edge;
+import org.gephi.graph.api.Graph;
+import org.gephi.graph.api.GraphModel;
+import org.gephi.graph.api.Node;
+import org.gephi.graph.api.Table;
+import org.gephi.io.exporter.spi.CharacterExporter;
+import org.gephi.io.exporter.spi.GraphExporter;
+import org.gephi.preview.types.EdgeColor;
+import org.gephi.project.api.Workspace;
+import org.gephi.utils.longtask.spi.LongTask;
+import org.gephi.utils.progress.Progress;
+import org.gephi.utils.progress.ProgressTicket;
+import uk.ac.ox.oii.jsonexporter.model.GraphEdge;
+import uk.ac.ox.oii.jsonexporter.model.GraphElement;
+import uk.ac.ox.oii.jsonexporter.model.GraphNode;
+
+public class JSONExporter implements GraphExporter, LongTask, CharacterExporter {
+
+    private boolean exportVisible = false;
+    private Workspace workspace;
+    private Writer writer;
+    private ProgressTicket progress;
+    private boolean cancel = false;
+    //private File path;
+
+    @Override
+    public boolean execute() {
+        try {
+            if (writer != null) {//path.getParentFile().exists()
+                GraphModel graphModel = workspace.getLookup().lookup(GraphModel.class);
+                Graph graph = null;
+                if (exportVisible) {
+                    graph = graphModel.getGraphVisible();
+                } else {
+                    graph = graphModel.getGraph();
+                }
+
```

Comment:
This exporter isn't resistant to a writer thread modifying the graph. The exporter graph might be inconsistent if another thread delete things between the nodes and edges export for instance. I suggest to add here the line `graph.readLock();` and then in the `finally {` block add `graph.readUnlock();`.